src/test/java/com/github_proxy2/api/GithubProxyApiTest.java

```diff
@@ -0,0 +1,165 @@
+@AutoConfigureWireMock(port = 8089)
+public class GithubProxyApiTest {
+
+    @Autowired
+    WireMockServer wireMockServer;
+
+    @Autowired
+    RestTemplate restTemplate;
+
+    @Autowired
+    RepoRepository repoRepository;
+
+    @Autowired
+    ObjectMapper mapper;
+
+    private static final String OWNER = "ABC";
+    private static final String REPO_NAME = "1234";
+
+    private static final GithubRepositoryDto GITHUB_REPO_DTO = new GithubRepositoryDto(
+            "pmarko09/school-supervision",
+            null, "https://github.com/pmarko09/school-supervision.git", 0,
+            LocalDateTime.of(2024, 9, 22, 16, 39, 38),
+            new Owner("abc"), "1234");
+
+    @Test
+    void getRepoFromGithubTest_DataCorrect_RepoDtoReturned() throws JsonProcessingException {
+        wireMockServer.stubFor(WireMock.get(WireMock.urlEqualTo(
+                        "/repos/" + OWNER + "/" + REPO_NAME))
+                .willReturn(WireMock.aResponse()
+                        .withStatus(HttpStatus.OK.value())
+                        .withHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                        .withBody(mapper.writeValueAsString(GITHUB_REPO_DTO))));
+
+        String url = UriComponentsBuilder.fromHttpUrl("http://localhost:8083")
+                .path("/repositories/" + OWNER + "/" + REPO_NAME)
+                .toUriString();
+
+        ResponseEntity<RepoDto> response = restTemplate.getForEntity(url, RepoDto.class);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        RepoDto result = response.getBody();
+        assertNotNull(result);
+        assertEquals(GITHUB_REPO_DTO.getFullName(), result.getFullName());
+        assertEquals(GITHUB_REPO_DTO.getDescription(), result.getDescription());
+        assertEquals(GITHUB_REPO_DTO.getStars(), result.getStars());
+        assertEquals(GITHUB_REPO_DTO.getCloneUrl(), result.getCloneUrl());
+        assertEquals(GITHUB_REPO_DTO.getCreatedAt(), result.getCreatedAt());
+    }
+
+    @Test
+    void saveRepoLocallyTest_DataCorrect_RepoDtoReturned() throws JsonProcessingException {
+        wireMockServer.stubFor(WireMock.get(WireMock.urlEqualTo(
+                        "/repos/" + OWNER + "/" + REPO_NAME))
+                .willReturn(WireMock.aResponse()
+                        .withStatus(HttpStatus.OK.value())
+                        .withHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
+                        .withBody(mapper.writeValueAsString(GITHUB_REPO_DTO))));
+
+        String url = UriComponentsBuilder.fromHttpUrl("http://localhost:8083")
+                .path("/repositories/" + OWNER + "/" + REPO_NAME)
+                .toUriString();
+
+        ResponseEntity<RepoDto> response = restTemplate.postForEntity(url, null, RepoDto.class);
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        RepoDto result = response.getBody();
+        assertNotNull(result);
```

Comment:
Zauważ, że jest to test integracyjny. Warto byłoby zweryfikować czy w bazie danych faktycznie zostały zapisane dane. 
Możemy wstrzyknąć repozytorium do naszej klasy testowej i wywołać find by id albo find by owner and repoName i sprwadzić czy repoztyroium faktycznie nam zwróci